### ISSUE TITLE
NVDA reports attachments for most messages on Outlook 365

### ISSUE THREAD

**Issue by jmdaweb** [nvaccess/nvda#15396]
### Steps to reproduce:

1. Open Outlook.
2. Navigate through the messages list using arrow keys.

### Actual behavior:

NVDA reports "attachment" on almost all list items, even in messages which only contain embedded images. These are not "real" attachment, and Outlook identifies them in a different way.

### Expected behavior:

NVDA should report attachments in messages with attached documents which aren't embedded in the message body.

#### NVDA installed/portable/running from source:
Installed and portable.
#### NVDA version:
2023.2
#### Windows version:
Windows 10 build 19045 and Windows 11 build 22621.
#### Name and version of other software in use when reproducing the issue:
Microsoft Outlook 365.
#### Does the issue still occur after restarting your computer?
Yes.
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Yes. 2023.1 and 2022.x, with same results.
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes.
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes.

**Comment by jmdaweb** [comment]
Follow-up of #4441. CC: @jmortizsilva

**Comment by CyrilleB79** [comment]
Confirmed on my side and it has been the case for years.

I disagree with the reason why #4441 was closed (see https://github.com/nvaccess/nvda/issues/4441#issuecomment-155327183).

Today, NVDA reports attachments in the message list for the two following cases:
1. A message having a real attachment. That is, when you are in the message body, you can shift+tab to attachments and then select them with arrows, open them with enter, copy them with control+C, etc.
2. A message with an image directly embedded in the message body. This can be a signature, a background image, the logo of a company included in the signature (very common case...)

Visually the attachment icon in the message list only occurs for case 1. (real attachments), not for case 2. (embedded images).

Unless there is a strong argument to have a different behaviour, the screen reader should follow what sighted people can see.

**Comment by Brian1Gaff** [comment]
Yes, this issue is not a new one, as going back to Outlook Express the same thing occurred, I always thought it was because of the annoying habit some people have of using graphic footers in their emails which used a non standard embedding  seen by email software as an  attachment.
 Brian

-- 
***@***.*** 
Sent via blueyonder.(Virgin media) 
Please address personal E-mail to:-
***@***.***, putting 'Brian Gaff'
in the display name field.

**Comment by Adriani90** [comment]
@CyrilleB79 I am not sure I fully agree with your statement. People were educated that attachment in Microsoft Outlook is something where you open the message and press alt+c or shift+tab and reach to some files.
However, an embeded image / object is an attachment as well since it opens in an external window when you access it. So actually NVDA behaves properly in this case.

But there is indeed one use case where NVDA should not report "has attachment" in the email list:
1. Open your email client
2. Open a new empty email to send
3. Paste a picture into it (e.g. a screenshot)
4. Send youself that email which contains only the picture

Actual: the email is announced with "has attachment" property when browsing through the list of emails

Expected: pictures pasted directly into the email body should not be considered as attachments, but rather as "has external content".

**Comment by CyrilleB79** [comment]
You can have any opinion on what should be reported as an attachment but:
1. For consistency and good communication with sighted people, NVDA should report has attachment when the attachment icon is visually displayed in the list.
2. If the attachment icon appears in the list in cases where it should not (or the contrary), this needs to be reported to Microsoft and discussed with them in the first place.

**Comment by Adriani90** [comment]
cc: @msftedad can you please report this to the development team?
Thanks.

**Comment by msftedad** [comment]
Hi @Adriani90, this issue is not reproducing on our end. If you are still experiencing the problem, please let us know.

**Version details tested:**

NVDA: 2025.3

Outlook: 1.2025.1114.300

**Comment by CyrilleB79** [comment]
I am able to reproduce. So msftedad should be able to reproduce, except if things have changed in very recent Office versions.

Anyway @Adriani90, the issue is not reproducible with Narrator. So NVDA may need to get the information otherwise than it currently does. Today, it uses object model to get this information.

Though Narrator is quite verbose since it reports "No attachments" when the message does not contain any, while reporting nothing would be more desirable. If NVDA gets the same information as Narrator, it may become verbose too, except if you request to MS to change what is reported. I have not checked if it's done via UIA.
